packages/gestalt/src/TableRow.js

```diff
@@ -1,10 +1,41 @@
 // @flow strict
-import React, { type Node } from 'react';
+import React, { Children, type Node, cloneElement, useContext, useEffect, useState } from 'react';
+import TableContext from './TableContextProvider.js';
 
 type Props = {|
   children: Node,
 |};
 
 export default function TableRow(props: Props): Node {
-  return <tr>{props.children}</tr>;
+  const { stickyColumns } = useContext(TableContext);
+  const rowRef = React.useRef();
+  const [columnWidths, setColumnWidths] = useState([]);
+
+  useEffect(() => {
+    if (rowRef && rowRef.current && stickyColumns) {
+      const colWidths = [];
+      const tableRowChildrenArray = [...rowRef.current.children];
+      tableRowChildrenArray.forEach((child) => {
+        colWidths.push(child.clientWidth);
+      });
+      setColumnWidths(colWidths);
+    }
+  }, [rowRef, stickyColumns]);
```

Comment:
is needed rowRef? rowRef.current aint needed. I'm not sure about this, just asking a question to discuss and learn.

Reply:
It isn't actually, this was left over from a previous attempt, good catch!

Reply:
https://github.com/facebook/react/issues/14387#issuecomment-503616820